pkg/controller/services.go

```diff
@@ -392,6 +418,60 @@ func apicDevCtx(name string, tenantName string,
 	return cc
 }
 
+func apicFilter(name string, tenantName string,
+	portSpec []v1.ServicePort) apicapi.ApicObject {
+
+	filter := apicapi.NewVzFilter(tenantName, name)
+	filterDn := filter.GetDn()
+
+	for i, port := range portSpec {
+		fe := apicapi.NewVzEntry(filterDn, strconv.Itoa(i))
+		fe.SetAttr("etherT", "ip")
+		if port.Protocol == v1.ProtocolUDP {
+			fe.SetAttr("prot", "udp")
+		} else {
+			fe.SetAttr("prot", "tcp")
+		}
+		pstr := strconv.Itoa(int(port.Port))
+		fe.SetAttr("sFromPort", pstr)
+		fe.SetAttr("sToPort", pstr)
+		filter.AddChild(fe)
+	}
+	return filter
+}
+
+func apicFilterSnat(name string, tenantName string,
+        portSpec []portRangeSnat) apicapi.ApicObject {
+
+        filter := apicapi.NewVzFilter(tenantName, name)
+        filterDn := filter.GetDn()
+
+        for i, port := range portSpec {
+
+                fe := apicapi.NewVzEntry(filterDn, strconv.Itoa(i))
```

Comment:
Move it into a function and pass the protocol as param to avoid repeated code here?

Reply:
Done, thanks!